fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -913,7 +978,23 @@ pub(crate) fn state_write_word<S: InterpreterStorage>(
         .merkle_contract_state_insert(contract, key, &value)
         .map_err(RuntimeError::Storage)?;
 
-    *exists = result.is_some() as Word;
+    *created_new = result.is_none() as Word;
+
+    if result.is_none() {
+        // New data was written, charge gas for it
+        let profiler = ProfileGas {
+            pc: pc.as_ref(),
+            is,
+            current_contract,
+            profiler,
+        };
+        gas_charge(
+            cgas,
+            ggas,
+            profiler,
+            (Bytes32::LEN as u64) * new_storage_gas_per_byte,
```

Comment:
I think we need to charge static cost and per byte=)

Reply:
What static cost? The opcode itself already costs for instertion, this is only for the new slot which has a fixed cost.

Reply:
It is related to the comment about `DepenentCost`=) `base` cost can include inserting the new entry into the storage(it will represent the burden of the SMT), and `deps_per_unit` will be gas per byte that you will multiply with `32` or `64`, but in the later versions with dynamic storage by the size of the value.